streamz.js

```diff
@@ -170,6 +170,10 @@ Streamz.prototype.end = function(d,cb) {
 Streamz.prototype.promise = function() {
   const buffer = [];
   const bufferStream = Streamz(d => {
+    if (this.options.maxBuffer && buffer.join('').length > this.options.maxBuffer) {
```

Comment:
It's probably better to track with a counter (i.e. `size += buffer.length || 1`). If stream is in `objectMode` then the counter has to be a simple increment (i.e. number of objects) as each object will not have `.length`